pkg/rules/rules.go

```diff
@@ -67,6 +80,52 @@ func (rr *GRPCClient) Rules(ctx context.Context, req *rulespb.RulesRequest) (*ru
 	return &rulespb.RuleGroups{Groups: resp.groups}, resp.warnings, nil
 }
 
+// filterRules filters rules in a group according to given matcherSets.
+func filterRules(ruleGroups []*rulespb.RuleGroup, matcherSets [][]*labels.Matcher) []*rulespb.RuleGroup {
+	if len(matcherSets) == 0 || len(ruleGroups) == 0 {
```

Comment:
Just a nit: Although there is no harm in having `len(ruleGroups) == 0` here, it is redundant, because the following loop wont run and it'll just return.

```suggestion
	if len(matcherSets) == 0 {
```

Reply:
Sure, removed it.